model_api/cpp/models/src/model_base.cpp

```diff
@@ -60,6 +60,18 @@ ModelBase::ModelBase(std::shared_ptr<ov::Model>& model, const ov::AnyMap& config
     inputsLayouts = parseLayoutString(layout);
 }
 
+void ModelBase::updateModelInfo() {
+    if (!model) {
+        std::runtime_error("The ov::Model object is not accessible");
+    }
+
+    if (!inputsLayouts.empty()) {
+        auto layouts = formatLayouts(inputsLayouts);
+        model->set_rt_info(layouts, "model_info", "layout");
```

Comment:
Not sure if we need this because we don't extract this info from IR. Plus MO itself can set these values in IR (not in rt_info)

Reply:
I think it is required to control the layout of input data so that the model could adjust to it and do preprocessing correctly.